koku/masu/external/downloader/azure/azure_report_downloader.py

```diff
@@ -58,17 +58,22 @@ def get_initial_dataframe_with_date(
     """
     dh = DateHelper()
     try:
-        data_frame = pd.read_csv(local_file, usecols=["UsageDateTime"])
+        data_frame = pd.read_csv(local_file, usecols=["UsageDateTime"], nrows=1)
         time_interval = "UsageDateTime"
         date_format = "%Y-%m-%d %H:%M:%S"
-        base_cols = utils.INGRESS_REQUIRED_COLUMNS
     except ValueError:
-        time_interval = "Date"
-        date_format = "%m/%d/%Y"
-        base_cols = utils.INGRESS_ALT_COLUMNS
-    use_cols = com_utils.fetch_optional_columns(local_file, base_cols, ["tags"], tracing_id, context)
-    data_frame = pd.read_csv(local_file, usecols=use_cols)
-    # Azure does not have an invoice column so we have to do some guessing here
+        try:
+            data_frame = pd.read_csv(local_file, usecols=["Date"], nrows=1)
+            time_interval = "Date"
+            date_format = "%Y-%m-%d"
+        except ValueError:
+            time_interval = "date"
+            date_format = "%m/%d/%Y"
+    # drop tag columns that only contain null values.
+    drop_cols = com_utils.list_null_optional_columns(local_file, ["tags"], tracing_id, context)
+    headers = [*pd.read_csv(local_file, nrows=1)]
+    data_frame = pd.read_csv(local_file, usecols=[c for c in headers if c not in drop_cols])
```

Comment:
Same thing as the AWS one, we're using pandas and reading the same CSV so many times here. I think we could cut down on how many CSV reads we are doing.

Reply:
Same thing